scripts/sleepSD.R

```diff
@@ -0,0 +1,96 @@
+source("scripts/fetch-data.R")
+
+participants <- 
+  arrow::open_dataset(
+    s3$path(stringr::str_subset(dataset_paths, "enrolledparticipants$"))
+  ) %>% 
+  select(c("ParticipantIdentifier",
+           "EnrollmentDate",
+           "CustomFields_InfectionFirstReportedDate")) %>% 
+  collect() %>% 
+  mutate(InfectionFirstReportedDate = as.Date(CustomFields_InfectionFirstReportedDate)) %>% 
+  select(-CustomFields_InfectionFirstReportedDate)
+
+fitbit_sleeplogs <- 
+  arrow::open_dataset(
+    s3$path(stringr::str_subset(dataset_paths, "sleeplogs$"))
+  )
+
+vars <- 
+  c("ParticipantIdentifier", 
+    "LogId",
+    "IsMainSleep",
+    "StartDate", # YYYY-MM-DDTHH:MM:SS format
+    "EndDate",
+    "Duration")
+
+sleeplogs_df <- 
+  fitbit_sleeplogs %>% 
+  select(all_of(c(vars))) %>% 
+  collect() %>% 
+  distinct() %>% 
+  mutate(
+    Date = lubridate::as_date(StartDate),
+    IsMainSleep = as.logical(IsMainSleep),
+    Duration = as.numeric(Duration),
+    SleepStartTime = lubridate::as_datetime(ifelse(IsMainSleep==TRUE, StartDate, NA)),
+    SleepEndTime = lubridate::as_datetime(ifelse(IsMainSleep==TRUE, EndDate, NA)),
+    MidSleep = format((lubridate::as_datetime(SleepStartTime) + ((Duration/1000)/2)), format = "%H:%M:%S"),
+    SleepStartTime = ((format(SleepStartTime, format = "%H:%M:%S") %>% lubridate::hms()) / lubridate::hours(24))*24,
+    SleepEndTime = ((format(SleepEndTime, format = "%H:%M:%S") %>% lubridate::hms()) / lubridate::hours(24))*24,
+    MidSleep = 24*lubridate::hms(MidSleep)/lubridate::hours(24)
+  ) %>% 
+  filter(IsMainSleep==TRUE)
+
+merged_data <- 
+  sleeplogs_df %>% 
+  left_join((participants %>% select(ParticipantIdentifier, InfectionFirstReportedDate)), 
+            by = "ParticipantIdentifier")
+
+# Weekly statistics
+weekly_stats <- 
+  list(
+    midsleep = 
+      merged_data %>%
+      group_by(ParticipantIdentifier, WeekStart = floor_date(Date, "week")) %>%
```

Comment:
Here you seem to be generating weekly stats, but only using weeks starting 3 months post-infection. You don't need to do that. You should only filter by 3-months post-infection for the timescale that uses all data starting 3-months post-infection.

Reply:
Resolved in [55dc440](https://github.com/Sage-Bionetworks/recover-sleep-measures/pull/2/commits/55dc440dd3683d9c6459679edc5d839f6f04b61f)